formats/json/commonMain/src/kotlinx/serialization/json/JsonElement.kt

```diff
@@ -76,13 +80,52 @@ public fun JsonPrimitive(value: String?): JsonPrimitive {
 @Suppress("FunctionName", "UNUSED_PARAMETER") // allows to call `JsonPrimitive(null)`
 public fun JsonPrimitive(value: Nothing?): JsonNull = JsonNull
 
+/**
+ * Create a [JsonPrimitive] from the given string, without surrounding it in quotes.
+ *
+ * **Note:** Creating a literal unquoted value of `null` (as in, `value == "null"`) is forbidden. If you want to create
+ * JSON null literal, use [JsonNull] object, otherwise, use [JsonPrimitive].
+ *
+ * This function is provided for encoding raw JSON values that cannot be encoded using the [JsonPrimitive] functions.
+ * For example,
+ *
+ * * precise numeric values (avoiding floating-point precision errors associated with [Double] and [Float]),
+ * * large numbers,
+ * * or complex JSON objects.
+ *
+ * Be aware that it is possible to create invalid JSON using this function!
+ *
+ * @see JsonPrimitive This is the preferred method for encoding JSON primitives.
+ * @throws JsonEncodingException If `value == "null"`
```

Comment:
```suggestion
 * @throws JsonEncodingException if `value == "null"`
```

because pop-ups in IDEA render these as the single sentence